File: pkg/db/models/oauth.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Oauth struct {
	ID            uuid.UUID `gorm:"primaryKey;type:uuid;default:gen_random_uuid()" json:"id"`
	UserID        uuid.UUID `gorm:"uniqueIndex:idx_userid_username" json:"userId"`
	User          User      `json:"-"`
	GitUserId     int32     `gorm:"not null" json:"gitUserId"`
	GitUsername   string    `gorm:"uniqueIndex:idx_userid_username" json:"username"`
	AccessToken   string    `json:"-"`
	GitProfileUrl string    `json:"url"`
	GitEmail      string    `json:"email"`
	GitName       string    `json:"name"`
	GitAvatarUrl  string    `json:"avatarUrl"`
	CreatedAt     time.Time `gorm:"autoCreateTime;column:created_at" json:"createdAt"`
	UpdatedAt     time.Time `gorm:"autoCreateTime;column:created_at" json:"updatedAt"`
}

File: web/src/features/select-repo/types/OAuth.ts
export interface IOAuth {
    'id': (string);
    'userId': string;
    'gitUserId': (number);
    'username': (string);
    'url': (string);
    'oauthType': (string);
    'email'?: (string);
    'name'?: (string);
    'avatarUrl'?: (string);
    'createdAt': Date;
    'updatedAt': Date;
}